libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx

```diff
@@ -42,22 +42,46 @@ export function SubscribeForm({
     resolver: yupResolver(schema),
   });
 
-  const onSubmit = useCallback(async (data: SubscribeFormFields) => {
-    try {
-      setSubscribeFormState(FormState.pending);
-      const response = await submitForm(data);
+  const onSubmitEmail = useCallback(async (form: SubscribeFormFields) => {
+    const request = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email: form.email,
+      }),
+    };
 
-      if (response.status === 200) {
-        setSubscribeFormState(FormState.success);
-      } else {
-        setSubscribeFormState(FormState.error);
-      }
+    try {
+      const response = await fetch('/api/sendgrid', request);
+      const data = await response.json();
+      return { status: response.status, data };
     } catch (error) {
       console.error(error);
-      setSubscribeFormState(FormState.error);
+      return { status: 500, data: error };
```

Comment:
revert all of this.
you need to just correctly handle response and error from `submitForm` method

Reply:
Done